tools/lsp/preview.rs

```diff
@@ -83,6 +88,140 @@ fn set_design_mode(enable: bool) {
     configure_design_mode(enable);
 }
 
+fn follow_child(element: &ElementRc) -> ElementRc {
+    let elem = element.borrow();
+    if elem.repeated.is_some() {
+        if let i_slint_compiler::langtype::ElementType::Component(base) = &elem.base_type {
+            return base.root_element.clone();
+        }
+    }
+    element.clone()
+}
+
+fn lsp_element_position(element: &ElementRc) -> (String, usize, usize, usize, usize) {
+    let e = &element.borrow();
+    e.node
+        .as_ref()
+        .and_then(|n| {
+            n.parent()
+                .filter(|p| p.kind() == i_slint_compiler::parser::SyntaxKind::SubElement)
+                .map_or_else(
+                    || Some(n.source_file.text_size_to_file_line_column(n.text_range().start())),
+                    |p| Some(p.source_file.text_size_to_file_line_column(p.text_range().start())),
+                )
+        })
+        .unwrap_or_else(|| (String::new(), 0, 0, 0, 0))
+}
+
+// triggered from the UI, running in UI thread
+pub fn select_element_at_impl(
+    x: f32,
+    y: f32,
+    component_instance: &ComponentInstance,
+    root_element: &ElementRc,
+) -> Option<ElementRc> {
+    let click_position = LogicalPoint::from_lengths(LogicalLength::new(x), LogicalLength::new(y));
+
+    for c in &root_element.borrow().children {
```

Comment:
looks like this function should recurse or something.
We want to get the most inner top element.